examples/tonic/src/server.rs

```diff
@@ -0,0 +1,96 @@
+use tonic::{transport::Server, Request, Response, Status};
+
+use hello_world::greeter_server::{Greeter, GreeterServer};
+use hello_world::{HelloReply, HelloRequest};
+use opentelemetry::api::{self, HttpTextFormat, KeyValue, Provider};
+use opentelemetry::sdk::Sampler;
+use opentelemetry::{global, sdk};
+use tracing_opentelemetry::{OpenTelemetryLayer, OpenTelemetrySpanExt};
+use tracing_subscriber::{Layer, Registry};
+
+pub mod hello_world {
+    tonic::include_proto!("helloworld"); // The string specified here must match the proto package name
+}
+
+#[derive(Debug, Default)]
+pub struct MyGreeter {}
+
+#[tonic::async_trait]
+impl Greeter for MyGreeter {
+    async fn say_hello(
+        &self,
+        request: Request<HelloRequest>, // Accept request of type HelloRequest
+    ) -> Result<Response<HelloReply>, Status> {
+        // Return an instance of type HelloReply
+        println!("Got a request: {:?}", request);
```

Comment:
take it or leave it: why not emit a tracing event here?

Reply:
Great point! I was just copying in the tutorial as an example so I hadn't really thought too hard about it